llvm/tools/llvm-profgen/PerfReader.cpp

```diff
@@ -41,6 +41,17 @@ static cl::opt<bool>
                                 "and produce context-insensitive profile."));
 cl::opt<bool> ShowDetailedWarning("show-detailed-warning",
                                   cl::desc("Show detailed warning message."));
+cl::opt<bool>
+    LeadingIPOnly("leading-ip-only",
```

Comment:
This name is a bit confusing. I think what you meant is to ignore LBRs and only consuming leading IPs? 

In that case, we should name it something like `ignore-lbr-samples`, to be consistent with existing `ignore-stack-samples`.

Reply:
Additionally, does use of `profiled-event` require `leading-ip-only`? If `leading-ip-only` is not specified, what is the expected behavior?